sdk/iot/provisioning/src/az_iot_provisioning_client_sas.c

```diff
@@ -46,24 +46,22 @@ AZ_NODISCARD az_result az_iot_provisioning_client_sas_get_signature(
   // Where
   // resource-string: <scope-id>/registrations/<registration-id>
 
-  int32_t required_size = 
-      az_span_size(client->_internal.id_scope)
-      + az_span_size(resources_string)
-      + az_span_size(client->_internal.registration_id)
-      + 1 // LF
-      + _az_iot_u32toa_size(token_expiration_epoch_time);
+  az_span remainder = signature;
+  int32_t signature_size = az_span_size(signature);
 
-  AZ_RETURN_IF_NOT_ENOUGH_SIZE(signature, required_size);
+  AZ_RETURN_IF_FAILED(_az_span_copy_url_encode(remainder, client->_internal.id_scope, &remainder));
 
-  az_span remainder = signature;
-  remainder = az_span_copy(remainder, client->_internal.id_scope);
+  AZ_RETURN_IF_NOT_ENOUGH_SIZE(signature, az_span_size(resources_string));
   remainder = az_span_copy(remainder, resources_string);
-  remainder = az_span_copy(remainder, client->_internal.registration_id);
+
+  AZ_RETURN_IF_FAILED(_az_span_copy_url_encode(remainder, client->_internal.registration_id, &remainder));
+  
+  AZ_RETURN_IF_NOT_ENOUGH_SIZE(signature, 1 /* LF */);
```

Comment:
?
```suggestion
  AZ_RETURN_IF_NOT_ENOUGH_SIZE(remainder, 1 /* LF */);
```

Reply:
Goodness, that's a big miss. Thanks for catching it.